Userland/Libraries/LibGL/Tex/Texture2D.cpp

```diff
@@ -18,11 +18,13 @@ void Texture2D::upload_texture_data(GLenum, GLint lod, GLint internal_format, GL
     // Considering we control this library, and `gl.h` itself, we don't need to add any
     // checks here to see if we support them; the program will simply fail to compile..
 
-    // Somebody passed us in nullptr...
-    // Apparently this allocates memory on the GPU (according to Khronos docs..)?
+    // Somebody passed us in nullptr... allocate uninitialized memory and set size
     if (pixels == nullptr) {
-        dbgln("LibGL: pixels == nullptr when uploading texture data.");
-        VERIFY_NOT_REACHED();
+        auto& mip = m_mipmaps[lod];
+        mip.set_width(width);
+        mip.set_height(height);
+        mip.pixel_data().resize(width * height);
+        return;
```

Comment:
This doesn't seem related to the commit it's in.

Reply:
Yes, I'll put it into a separate one